code/game/machinery/iv_drip.dm

```diff
@@ -126,8 +126,11 @@
 		return PROCESS_KILL
 
 	if(!(get_dist(src, attached) <= 1 && isturf(attached.loc)))
-		to_chat(attached, span_userdanger("The IV drip needle is ripped out of you!"))
-		attached.apply_damage(3, BRUTE, pick(BODY_ZONE_R_ARM, BODY_ZONE_L_ARM))
+		to_chat(attached, span_userdanger("The IV drip needle is ripped out of you, leaving an open bleeding wound!"))
+		var/selected_limb = pick(BODY_ZONE_R_ARM, BODY_ZONE_L_ARM)
+		var/obj/item/bodypart/chosen_limb = attached.get_bodypart(selected_limb)
+		attached.apply_damage(3, BRUTE, selected_limb)
+		chosen_limb.force_wound_upwards(/datum/wound/pierce/moderate)
```

Comment:
```suggestion
		var/list/arm_zones = shuffle(list(BODY_ZONE_R_ARM, BODY_ZONE_L_ARM))
		var/obj/item/bodypart/chosen_limb = attached.get_bodypart(arm_zones[1]) || attached.get_bodypart(arm_zones[2]) || attached.get_bodypart(BODY_ZONE_CHEST)
		chosen_limb.receive_damage(3)
		chosen_limb.force_wound_upwards(/datum/wound/pierce/moderate)
```

I feel like I'm super overthinking this, but this is the smoothest way I can think of to prioritize both arms and fallback to the chest if they're armless. It won't account for if the randomly picked limb is augmented, but at least then the worst that can happen is they don't take the wound instead of the code runtiming because the arm doesn't exist.

Reply:
Oh I missed the actual code cause I was looking on mobile earlier, this is way better, thank you